File: app/src/main/java/dev/blackcat/minauta/net/ConnectionFactory.java
package dev.blackcat.minauta.net;

public class ConnectionFactory
{

    public enum Type
    {
        FAKE,
        JNAUTA
    }

    public static <T extends Connection> T createSessionProducer(Class<T> clazz)
    {
        try
        {
            T instance = clazz.newInstance();
            return instance;
        }
        catch (Exception e)
        {
            e.printStackTrace();
        }
        return null;
    }
}

File: app/src/main/java/dev/blackcat/minauta/net/ConnectionFactory.kt
package dev.blackcat.minauta.net

object ConnectionFactory {

    enum class Type {
        JNAUTA,
        FAKE,
        FAKE_NO_LOGIN,
        FAKE_NO_LOGOUT
    }

    fun create(type: Type): Connection {
        return when (type) {
            Type.JNAUTA -> { JNautaConnection::class.java.newInstance() }
            Type.FAKE -> { FakeConnection::class.java.newInstance() }
            Type.FAKE_NO_LOGIN -> { FakeNoLoginConnection::class.java.newInstance() }
            Type.FAKE_NO_LOGOUT -> { FakeNoLogoutConnection::class.java.newInstance() }
        }
    }

    fun create(): Connection {
        return create(Type.FAKE)
    }
}
